contrib/epee/src/net_ssl.cpp

```diff
@@ -558,3 +565,36 @@ bool ssl_support_from_string(ssl_support_t &ssl, boost::string_ref s)
 } // namespace
 } // namespace
 
+#ifdef _WIN32
+
+// https://stackoverflow.com/questions/40307541
+// Because Windows always has to do things wonkily
+#include <wincrypt.h>
+static void add_windows_root_certs(boost::asio::ssl::context &ctx)
```

Comment:
Are these all non-throwing C functions? Can this be made `noexcept`? If not, we have to worry about throwing + memory leaks.

Reply:
Mostly., but it uses a boost context, so it can probably throw in some unlikely out of memory cases. I'll catch and call set_default_paths in that case. The caller can throw already anyway.

Reply:
The caller isn't an issue, its a potential leak if one of the calls in this function throws. Also, the function could be given the context pointer directly.